velox/common/memory/MmapAllocator.cpp

```diff
@@ -263,6 +263,8 @@ bool MmapAllocator::allocateContiguousImpl(
   const auto totalCollateralPages =
       numCollateralPages + numLargeCollateralPages;
   if (numPages == 0) {
+    numAllocated_ -= totalCollateralPages;
+    numMapped_ -= numLargeCollateralPages;
```

Comment:
the `numExternalMapped_` also needs to be updated here.

Actually can you move this check to the top of the method and do :

```
  if (numPages == 0) {
    freeNonContiguous(*collateral);
    freeContiguous(allocation);
    return;
  }
```

This should take care of the accounting correctly.